### ISSUE TITLE
Shadowsocks Android bug

### ISSUE THREAD

**Issue by KarlM256** [shadowsocks/shadowsocks-android#345]
Hi All...
There is a bug in SS for Android version 2.7.5.

When the SS on/off button is turned to the on position and the desired server is not running, the SS Android on/off button will often turn green, showing a successful connection. A successful connection is not possible because the server is not running. Occasionally, SS Android detects that the server is unavailable and reports an error.

This happens with version 2.7.5 and several earlier versions.

App settings:
  Server specified by a DDNS address
  Port 443
  Local Port 1080
  Password ...
  Encryption Method AES-256-CFB
  Route All
  Global Proxy Yes
  UDP Forwarding No
  Auto Connect No

The easiest way to demonstrate this bug is to take a working client and change the Port to an unused value. Then try to connect, and see that the App shows success when it should not show success.

By the way, thanks for fixing the reset bug with the latest update.

Thanks,

...Karl

**Comment by madeye** [comment]
Update to 2.7.7

**Comment by KarlM256** [comment]
Hi All...

![screenshot_2015-08-17-16-45-03](https://cloud.githubusercontent.com/assets/13839705/9319205/ac3d2254-4500-11e5-8f1b-ddde6800b182.jpg)

I am running 2.7.7 and the problem persists. I have attached a screenshot to show the issue. I have connected to a nonexistent shadowsocks server, and the apk shows success. But when I run Firefox for example, it hangs. If I connect it to my personal server, it works fine. Also, if I try to connect to an invalid DNS address, shadowsocks briefly displays an error message about the connection, but then hangs for a very long time. I am still running on a Verizon LG G2, running Android 5.0.2, and I also have the same problem on a Nexus 9, running Android 5.1.1 with the stagefright update.

Thanks,

...Karl
